### PR TITLE
:cheer: Minimum Viable Product

### LINKED ISSUE #3: Handle cancellation token errors.
When a request is 'cut' mid way before the response is fully sent AND the action method handled was an `async` action method, some crap is recorded in the log files.

I asked Andrew to help suggest something after providing some repo code, etc. and [he blogged about it](https://andrewlock.net/using-cancellationtokens-in-asp-net-core-mvc-controllers/).

So implement this custom catch code.

### LINKED ISSUE #4: Add NuGet
So we (and others!) can make use of this awesome sauce 😎

We'll need to setup one of those public NuGet org accounts.

### LINKED ISSUE #7: SourceLink enable the nuget package
REF: https://www.hanselman.com/blog/ExploringNETCoresSourceLinkSteppingIntoTheSourceCodeOfNuGetPackagesYouDontOwn.aspx

### LINKED ISSUE #9: Common HomeController
Have a common home controller that can be used and configured.

- ability to set the 'banner'
- ability to set this up in `startup.cs` so the banner can be configured. Also, helps with not having to rely on reflection for auto-finding the route.

### PR DESCRIPTION
**Pull request by PureKrome**
- Fixes #3
- Fixes #4 
- Fixes #7 
- Fixes #9

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,6 +13,8 @@ This library contains a collection of helpers, models and extension methods that
 - [Consistent Api Error schema.](#Sample1)
```
Comment (RPM1984) [inline README.md:13]: Wiki needs some more updates.
- Update to include problem JSON
- Remove FluentValidation stuff
- Some JSON exception stuff needs to be removed

File: src/Homely.AspNetCore.Mvc.Helpers/Controllers/HomeController.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
+﻿using Homely.AspNetCore.Mvc.Helpers.Models;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using System;
+
+namespace Homely.AspNetCore.Mvc.Helpers.Controllers
+{
+    [AllowAnonymous]
```
Comment (RPM1984) [inline src/Homely.AspNetCore.Mvc.Helpers/Controllers/HomeController.cs]: should this be removed? 

actually.. is this controller needed, here? shouldn't it be in the test application / template?

File: tests/Homely.AspNetCore.Mvc.Helpers.Tests/TestControllerTests/ErrorTests.cs [hunk 3]
```diff
@@ -13,127 +9,50 @@ namespace Homely.AspNetCore.Mvc.Helpers.Tests.TestControllerTests
...
-            apiErrorResult.StackTrace.ShouldNotBeNullOrWhiteSpace();
+            await response.Content.ShouldLookLike(error);
         }
 
-        [Fact]
-        public async Task GivenAnAjaxGetRequestWithCorsEnabled_Error_ReturnsAnHttp500()
-        {
-            // Arrange.
+        //[Fact]
+        //public async Task GivenAGetRequestAndIncludeStackTrace_Error_ReturnsAnHttp500()
```
Comment (RPM1984) [inline tests/Homely.AspNetCore.Mvc.Helpers.Tests/TestControllerTests/ErrorTests.cs]: can we nuke this test? or should be uncommented?

**Review by RPM1984 — CHANGES_REQUESTED** [review]
Looks good! Just the wiki updates + question about HomeController

File: README.md (hunk 1 above)
Comment (PureKrome) [reply README.md:13]: This is slated for milestone 1.1 as mentioned in #8

File: src/Homely.AspNetCore.Mvc.Helpers/Controllers/HomeController.cs (hunk 2 above)
Comment (PureKrome) [reply src/Homely.AspNetCore.Mvc.Helpers/Controllers/HomeController.cs]: > should this be removed?
Yep. 🔧 

> actually.. is this controller needed, here?
totally.

```
service.AddMvcCore( .. )
       .AddAHomeController(services, typeof(Startup), "pew pew")...
```

File: tests/Homely.AspNetCore.Mvc.Helpers.Tests/TestControllerTests/ErrorTests.cs (hunk 3 above)
Comment (PureKrome) [reply tests/Homely.AspNetCore.Mvc.Helpers.Tests/TestControllerTests/ErrorTests.cs]: 🔧 I couldn't figure out a way to set some Env settings for the test host. So I can do that later, if/when I figure that out.

**Comment by PureKrome** [comment]
👀 @RPM1984

**Review by RPM1984 — APPROVED** [review]

**Pull request merged**